bundleserver/src/test/java/net/discdd/server/ADUEnd2EndTest.java

```diff
@@ -0,0 +1,63 @@
+package net.discdd.server;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.whispersystems.libsignal.IdentityKey;
+import org.whispersystems.libsignal.IdentityKeyPair;
+import org.whispersystems.libsignal.ecc.Curve;
+import org.whispersystems.libsignal.ecc.ECKeyPair;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Base64;
+
+import static net.discdd.server.ADUEnd2EndTest.DDDPEMEncoder.ECPrivateKeyType;
+import static net.discdd.server.ADUEnd2EndTest.DDDPEMEncoder.ECPublicKeyType;
+
+@SpringBootTest
+public class ADUEnd2EndTest {
+    @TempDir
+    static Path tempRootDir;
+
+    /**
+     * This class is a bit funky. It uses URL base64 encoding instead of the standard base64 encoding.
+     */
+    public class DDDPEMEncoder {
+
+        private static final String KEY_HEADER = "-----BEGIN %s KEY-----\n";
+        private static final String KEY_FOOTER = "\n-----END %s KEY-----";
+
+        public static final String ECPrivateKeyType = "EC PRIVATE";
+        public static final String ECPublicKeyType = "EC PUBLIC";
+
+        public static String encode(byte[] bytes, String type) {
```

Comment:
this'd better be kept in a testUtils class if it can possibly be used by some other tests too

Reply:
good point. i've moved it to bundle-core net.discdd.bundleserver